### PR TITLE
feat(login): add support of localStorage for user registration

### PR DESCRIPTION
**Pull request by amitgandole**
- added custom hook for storing and retrieving localStorage value
- added support for storing user info in localStorage
- corrected reducer implementation

### CODE REVIEW

File: src/components/Login/Login.tsx [hunk 1]
```diff
@@ -1,16 +1,71 @@
...
+// Custom hooks
+import useLocalStorage from "../../utils/useLocalStorage";
+
+// Css
+import "./Login.css";
+import generateUniqueId from "../../utils/useUniqueId";
 
 const { Option } = Select;
 
 const Login: React.FC = () => {
```
Comment (pranitkhadilkar7) [inline src/components/Login/Login.tsx:23]: React.FC is not required

File: src/utils/useLocalStorage.ts
Comment (pranitkhadilkar7) [inline src/utils/useLocalStorage.ts:1]: You don't have to use custom hook to access localstorage. I would suggest clear some reusable functions to get and set data into local storage and use that

File: src/reducers/authReducer.ts [hunk 2]
```diff
@@ -13,7 +13,11 @@ const authSlice = createSlice({
   initialState,
   reducers: {
     login: (state, action: PayloadAction<User>) => {
-      state = action.payload;
+      const { id, firstName, lastName, role } = action.payload;
```
Comment (pranitkhadilkar7) [inline src/reducers/authReducer.ts:16]: may be you can use state.user to set the user object here

File: src/utils/useLocalStorage.ts
Comment (amitgandole) [reply src/utils/useLocalStorage.ts:1]: Needed this hook for further localStorage operations.

File: src/components/Login/Login.tsx (hunk 1 above)
Comment (amitgandole) [reply src/components/Login/Login.tsx]: Done

**Review by pranitkhadilkar7 — APPROVED** [review]

**Pull request merged**